linux_os/guide/system/auditing/auditd_configure_rules/audit_unsuccessful_file_modification/audit_rules_unsuccessful_file_modification_detailed_open/oval/shared.xml

```diff
@@ -0,0 +1,596 @@
+<def-group>
```

Comment:
Is it me or is this OVAL unnecessarily complicated? Hard to tell by the comments in the file.

Reply:
> Hard to tell by the comments in the file.

Or lack of comments?

First there are  `constant_variables` definitions used to build up the rules.
Then, the local variables put the defined constants, and some other data together to form the rules to look for.
After that, a ton of objects that must exist, one for each rule.
And a test to check order of the rules that were retrieved.

This is repeated for each arch (b32, b64, error code (EACCES, EPERM), and audit daemon (augenrules, auditctl).